torch/_refs/__init__.py

```diff
@@ -596,6 +597,11 @@ def isneginf(a: TensorLikeType) -> TensorLikeType:
 def isnan(a: TensorLikeType) -> TensorLikeType:
     return prims.ne(a, a)
 
+@_make_elementwise_unary_reference(ELEMENTWISE_TYPE_PROMOTION_KIND.ALWAYS_BOOL)
+def isreal(a: TensorLikeType) -> TensorLikeType:
+    if utils.is_complex_dtype(a.dtype):
+        return prims.imag(a) == 0
```

Comment:
After primTorch started a context manager, TorchRefsMode, was added that can remap torch operations into their Python references. Because of this our new standard is to call the torch operation when available, and only directly call another ref or prim if that ref or prim has additional functionality not supported by the torch operation. For example, elementwise prims should always accept tensors or scalars, but not every elementwise torch operation accepts scalar arguments. 

Following that, in this case I think `torch.imag()` is the thing.